validation/validation.go

```diff
@@ -0,0 +1,370 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+)
+
+// New is the main function in charge of validating the HTTP request form by using the defined rule set.
+// Form fields will be validated only if there is a rule that indicates they must be validated.
+func New(form url.Values, ruleSet ...Rule) map[string][]string {
+	verrs := make(map[string][]string)
+	mutex := new(sync.RWMutex)
```

Comment:
I don't think this is needed @tatang26. I say this because the variable you're locking on is specific to the function scope and goes passes by value.